specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,567 @@
+}
+
+// Send subsequent progress updates
+winrt::Windows::Foundation::IAsyncAction UpdateProgressAsync()
+{
+    winrt::hstring tag = L"weekly-playlist";
+    winrt::hstring group = L"downloads";
+
+    // Assign new values
+    // Note that you only need to assign values that changed. In this example we don't assign progressStatus since we don't need to change it
+    AppNotificationProgressData data;
+    data.Value(0.7); // Binds to {progressValue} in xml payload
+    data.ValueStringOverride(L"18/26 songs"); // Binds to {progressValueString} in xml payload
+    data.SequenceNumber(2);
+
+    auto result = co_await AppNotificationManager::Default().UpdateProgressDataAsync(data, tag, group);
+    if (result != AppNotificationProgressResult::Succeeded)
+    {
+        LOG_HR_MSG(E_UNEXPECTED, "Notification Progress Update Failed!");
+    }
+}
+```
+# Remarks
+
+## Registration
+
+The developer should always call the Notification Registration API first to register the current process as
+the Activator target.
+
+## Foreground API calls
+
+The developer should always subscribe to AppNotification activation foreground events to intercept notification related actions if
+the app happens to be running in the foreground.
+
+## Manifest Registration
+
+For MSIX, the COM activator GUID and the exe need to be registered in the manifest. The launch args
+would need to be pre-set to a well-known string that defines Activation Triggers.
+
+```xml
+<Extensions>
+  <com:Extension Category="windows.comServer">
+    <com:ComServer>
+      <com:ExeServer Executable="SampleBackgroundApp\SampleBackgroundApp.exe" DisplayName="SampleBackgroundApp" Arguments="----AppNotificationActivationServer">
+        <com:Class Id="{GUIDEntryPoint}" />
+      </com:ExeServer>
+    </com:ComServer>
+  </com:Extension>
+</Extensions>
+```
+
+# API Details
+
+```c#
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Windows.AppNotifications
+{
+    [experimental]
+    // The Shell asset details for Unpackaged App Registrations 
+    runtimeclass AppNotificationAssets
+    {
+        // Initialize using Shell assets like DisplayName and iconPath
+        AppNotificationAssets(String displayName, Windows.Foundation.Uri iconUri);
+
+        // The App friendly DisplayName for the Notification in Action Centre 
+        String DisplayName { get; };
+
+        // The full file path for the icon image
+        Windows.Foundation.Uri IconUri { get; };
+    }
+
+    [experimental]
+    // The Registration Info for Packaged and Unpackaged Notification Activations
+    runtimeclass AppNotificationActivationInfo
+    {
+        // Initialize using a manifest defined COM Activator Id. Only applicable to Packaged Win32 applications
+        static AppNotificationActivationInfo CreateFromActivationGuid(Guid taskClsid);
```

Comment:
the clsid is a COM local server CLSID. seems those terms should be used for the factory function, or just use a constructor to avoid having to give this a name.

Reply:
The issue here is that we would need 2 constructors: One that takes in a CLSID and another that takes in AppNotificationAssets. Overload constructors are disallowed in midl3 the last I checked.

Reply:
Added 2 ctors. This happens to work now in the idl because the 2 ctors have different number of arguments.